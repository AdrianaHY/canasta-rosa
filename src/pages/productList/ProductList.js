import React, { Component } from 'react';

import Card from '../../components/card/Card';

import Banner from '../../assets/banner.png';

import './ProductList.scss';

class ProductList extends Component {
	render() {
		const { products, isLoading } = this.props;
		return(
			<div className="product-list-container">
				{ isLoading &&
					<div className="lds-heart">
						<div className="lds-heart-container">
							<div></div>
						</div>
					</div>
				}
				<div className="image-container">
					<img src={Banner}  alt="banner" className="banner"/>
				</div>
				<ul className="product-list">
					{products.map((product) =>
						<Card product={product} key={product.name}/>
					)}
				</ul>
			</div>
		)
	}
}

export default ProductList;