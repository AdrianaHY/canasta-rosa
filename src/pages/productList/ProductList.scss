@import '../../styles/_variables.scss';

.product-list-container {
	.lds-heart {
		display: flex;
		position: relative;
		justify-content: center;
		align-items: center;
		width: 100vw;
		height: 100vh;
		transform-origin: 40px 40px;
		background-color: $lightPink;

		.lds-heart-container {
			position: relative;
			width: 80px;
			height: 80px;
			transform: rotate(45deg);
			transform-origin: 40px 40px;

			div {
				top: 32px;
				left: 32px;
				position: absolute;
				width: 32px;
				height: 32px;
				background: $white;
				animation: lds-heart 1.2s infinite cubic-bezier(0.215, 0.61, 0.355, 1);

				&::after,
				&::before {
					content: " ";
					position: absolute;
					display: block;
					width: 32px;
					height: 32px;
					background: $white;
				}

				&:before {
					left: -24px;
					border-radius: 50% 0 0 50%;
				}

				&:after {
					top: -24px;
					border-radius: 50% 50% 0 0;
				}
			}
		}

		@keyframes lds-heart {
			0% {
			  transform: scale(0.95);
			}
			5% {
			  transform: scale(1.1);
			}
			39% {
			  transform: scale(0.85);
			}
			45% {
			  transform: scale(1);
			}
			60% {
			  transform: scale(0.95);
			}
			100% {
			  transform: scale(0.9);
			}
		}

	}

	.image-container {
		display: flex;
		justify-content: center;
		padding: $paddingLarge;
	}

	.banner {
		width: 100%;

		@include breakpoint('large') {
			width: 50%;
		}
	}

	.product-list {
		padding-top: 0;
		display: flex;
		justify-content: center;
		flex-direction: row;
		flex-wrap: wrap;
		width: 100vw;
		margin-block-start: 0;
		margin-block-end: 0;
		margin-inline-start: auto;
		padding-inline-start: 0;

		@include breakpoint ('large') {
			padding: 0;
			padding-top: $paddingLarge;
		}
	}
}