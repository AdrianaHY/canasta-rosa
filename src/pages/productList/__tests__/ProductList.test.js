import React from 'react';
import { shallow } from 'enzyme';
import ProductList from '../ProductList';

describe('pages | productList | ProductList', () => {
	it('Renders the component', () => {
		const props = {
			products: [
				{
					name: 'product'
				}
			],
			isLoading: false,
			handlePage: jest.fn()
		};

		const component = shallow(
			<ProductList { ...props } />
		);

		expect(component).toMatchSnapshot();

		component.setProps({ isLoading: true });

		expect(component.find('.lds-heart')).toHaveLength(1);

	})
})