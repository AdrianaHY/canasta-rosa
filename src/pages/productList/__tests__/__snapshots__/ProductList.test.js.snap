// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages | productList | ProductList Renders the component 1`] = `ShallowWrapper {}`;
